workers/event_catcher/event_catcher.rb

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env ruby
+
+require "manageiq-messaging"
+require "pathname"
+require "sd_notify"
+
+class EventCatcher
+  def initialize(ems_id, default_endpoint, default_authentication, messaging_opts, page_size = 20)
+    @ems_id         = ems_id
+    @hostname       = default_endpoint["hostname"]
+    @username       = default_authentication["userid"]
+    @password       = default_authentication["password"]
+    @port           = default_endpoint["port"]
+    @messaging_host = messaging_opts["host"]
+    @messaging_port = messaging_opts["port"]
+    @page_size      = page_size
+  end
+
+  def run!
+    vim                     = connect
+    event_history_collector = create_event_history_collector(vim, page_size)
+    property_filter         = create_property_filter(vim, event_history_collector)
+
+    notify_started
+
+    wait_for_updates(vim) do |property_change|
+      next unless property_change.name.match?(/latestPage.*/)
+
+      events = Array(property_change.val).map { |event| parse_event(event) }
+      publish_events(events)
+    end
+  rescue Interrupt
+    # Catch SIGINT
```

Comment:
Will SIGKILL go through here as well? Typically you don't want to rescue that one

Reply:
No just SIGINT

Reply:
SIGTERM already worked without raising an exception, this just prevents ctrl-c from blowing up in an ugly way